src/os/windows/win32.cpp

```diff
@@ -562,10 +562,10 @@ std::wstring OTTD2FS(const std::string &name)
 	int name_len = (name.length() >= INT_MAX) ? INT_MAX : (int)name.length();
 	int len = MultiByteToWideChar(CP_UTF8, 0, name.c_str(), name_len, nullptr, 0);
 	if (len <= 0) return std::wstring();
-	wchar_t *system_buf = AllocaM(wchar_t, len + 1);
+	std::wstring system_buf(len + 1, L'\0');
```

Comment:
```suggestion
	std::wstring system_buf(len, L'\0');
```
MultiByteToWideChar() returned length includes terminating null character.

The old code used +1 for extra safety but cropped the result on return.